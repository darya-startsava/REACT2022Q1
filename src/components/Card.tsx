import CardType from '../types/card';

export default function Card({ id, image, name, year, director, actors, imdb, oscars }: CardType) {
  return (
    <div>
      <img src={`posters/${image}`} width="200" height="auto" alt="poster" />
      <h5>&ldquo;{name}&rdquo;</h5>
      <p>{year} year</p>
      <p>Director: {director}</p>
      <p>Starring: {actors}</p>
      <p>IMDb: {imdb}/10</p>
      <p>Oscars: {oscars}</p>
    </div>
  );
}
