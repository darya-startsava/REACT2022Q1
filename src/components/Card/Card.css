.card-image {
  width: 200px;
  height: auto;
}

.card-item {
  width: 200px;
  height: auto;
}

.close-img {
  width: 20px;
  height: 20px;
}

.close-button {
  border: none;
  background-color: #fff;
  margin-left: 180px;
}