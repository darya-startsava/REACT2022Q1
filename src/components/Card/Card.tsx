import './Card.css';
import CardType from '../../types/card';

export default function Card({
  id,
  image,
  uploadedImage,
  name,
  year,
  director,
  actors,
  imdb,
  oscars,
  gender,
  dateOfBirth,
  countryOfBirth,
  movieGenres,
  overview,
  releaseDate,
  voteAverage,
  isFull,
}: CardType) {
  let style = {};
  if (isFull) {
    style = { width: 500 };
  }
  return (
    <div className="my-1 d-flex flex-column card-item" style={style} data-testid="card">
      {image && <img src={image} className="align-self-center card-image" alt="poster" />}
      {uploadedImage && <img src={uploadedImage} className="card-image" alt="avatar" />}
      <h5 className="align-self-center card-title">{name}</h5>
      {year && <p>{year} year</p>}
      {director && <p>Director: {director}</p>}
      {actors && <p>Starring: {actors ? actors.join(', ') : ''}</p>}
      {imdb && <p>IMDb: {imdb}/10</p>}
      {oscars && <p>Oscars: {oscars}</p>}
      {gender && <p>{gender}</p>}
      {dateOfBirth && <p>Date of birth: {dateOfBirth}</p>}
      {countryOfBirth && <p>Country of birth: {countryOfBirth}</p>}
      {movieGenres && <p>Genres: {movieGenres ? movieGenres.join(', ') : ''}</p>}
      {isFull && <p className="overview px-2">Overview: {overview}</p>}
      {isFull && <p className="px-2">Release date: {releaseDate}</p>}
      {isFull && <p className="px-2">Vote average: {voteAverage}</p>}
    </div>
  );
}
