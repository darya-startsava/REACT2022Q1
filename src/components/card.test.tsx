import { render } from '@testing-library/react';

import Card from './card';

describe('Card', () => {
  test('information about director', () => {
    const { container } = render(
      <Card
        id={3}
        image="inception.jpg"
        name="Inception"
        year={2010}
        director="Christopher Nolan"
        actors={[
          'Leonardo DiCaprio',
          'Ken Watanabe',
          'Joseph Gordon-Levitt',
          'Marion Cotillard',
          'Elliot Page',
          'Tom Hardy',
          'Cillian Murphy',
          'Tom Berenger',
          'Michael Caine',
        ]}
        imdb={8.8}
        oscars={4}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
