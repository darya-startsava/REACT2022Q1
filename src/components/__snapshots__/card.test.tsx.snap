// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card information about director 1`] = `
<div>
  <div>
    <img
      alt="poster"
      height="auto"
      src="posters/inception.jpg"
      width="200"
    />
    <h5>
      “
      Inception
      ”
    </h5>
    <p>
      2010
       year
    </p>
    <p>
      Director: 
      Christopher Nolan
    </p>
    <p>
      Starring: 
      Leonardo DiCaprio
      Ken Watanabe
      Joseph Gordon-Levitt
      Marion Cotillard
      Elliot Page
      Tom Hardy
      Cillian Murphy
      Tom Berenger
      Michael Caine
    </p>
    <p>
      IMDb: 
      8.8
      /10
    </p>
    <p>
      Oscar: 
      4
    </p>
  </div>
</div>
`;
